pkg/network/setup/netpod/state_test.go

```diff
@@ -104,10 +115,27 @@ var _ = Describe("state", func() {
 		cache.stateCache[nets[5].Name] = netcache.PodIfaceNetworkPreparationFinished
 
 		state := netpod.NewState(cache, nil)
-		pending, started, err := state.PendingAndStarted(nets)
+		pending, started, finished, err := state.PendingStartedFinished(nets)
 
 		Expect(err).NotTo(HaveOccurred())
 		Expect(pending).To(Equal([]v1.Network{nets[0], nets[1]}))
 		Expect(started).To(Equal([]v1.Network{nets[2], nets[3]}))
+		Expect(finished).To(Equal([]v1.Network{nets[4], nets[5]}))
+	})
+
+	It("succeeds deleting network state", func() {
+		state := netpod.NewState(newConfigStateCacheStub(), nil)
+		nets := []v1.Network{{Name: netName}}
+		Expect(state.SetFinished(nets)).To(Succeed())
+
+		Expect(state.Delete(nets)).To(Succeed())
+
+		pending, started, finished, err := state.PendingStartedFinished(nets)
+		Expect(err).NotTo(HaveOccurred())
+
+		// On deletion, all networks cache are initialized back to "pending".
```

Comment:
Does the cache removed entirely when the interface is no longer exist in the spec? (removed by the controller following uplug)